components/domains/registerV2.tsx

```diff
@@ -150,6 +158,12 @@ const RegisterV2: FunctionComponent<RegisterV2Props> = ({ domain, groups }) => {
     }
   }, [priceData, priceError, duration, domain]);
 
+  useEffect(() => {
+    if (renewPriceError || !renewPriceData) return;
+    const high = renewPriceData?.["price"].high << BigInt(128);
+    setRenewPrice((renewPriceData?.["price"].low + high).toString(10));
+  }, [renewPriceData, renewPriceError, duration, domain]);
```

Comment:
No need to add another call to the RPC here. 

Please simply use `priceData` and `getPriceFromDomain` in the useEffect line 151.

1/ You can set `getPriceFromDomain(1, domain)`
2/ You can divide the priceData by the duration on the `else`